python/cudf/dataframe/dataframe.py

```diff
@@ -2383,54 +2399,24 @@ def __init__(self, df):
         self._df = df
 
     def __getitem__(self, arg):
-        row_slice = None
-        row_label = None
-
-        if isinstance(arg, int):
-            if arg < 0 or arg >= len(self._df):
-                raise IndexError("label scalar %s is out of bound" % arg)
-            row_label = arg
-            col_slice = self._df.columns
-
-        elif isinstance(arg, tuple):
-            arg_1, arg_2 = arg
-            if isinstance(arg_1, int):
-                row_label = arg_1
-            elif isinstance(arg_1, slice):
-                row_slice = arg_1
-            else:
-                raise TypeError(type(arg_1))
-            col_slice = arg_2
-
+        if isinstance(arg, tuple):
+            row_slice, col_slice = arg
+            if isinstance(col_slice, pd.Series):
+                # equivalent of core.indexing._getitem_iterable
+                df = pd.DataFrame(index=self._df.index)
+                return df
```

Comment:
I'm not following the logic here. You're returning an empty pandas DataFrame with the index without using the input tuple. Could you explain this?

Reply:
This should be gone now.  I was doing this while trying to trace some functionality around select_dtypes